File: Problems/0167.Two-Sum-II-Input-Array-Is-Sorted/TwoSum2InputArrayIsSorted.py
from typing import List

class Solution:

    # Runtime: 56 ms, faster than 98.27% of Python3 online submissions
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        rec = {}
        for i, n in enumerate(numbers):
            if n in rec:
                return [rec[n] + 1, i + 1]
            else:
                key = target - n
                rec[key] = i
        return []

File: leetcode/0167.Two-Sum-II-Input-Array-Is-Sorted/TwoSum2InputArrayIsSorted.go
package leetcode

// Runtime: 4 ms, faster than 96.00% of Go online submissions
func twoSum(numbers []int, target int) []int {
	rec := make(map[int]int)
	for i, n := range numbers {
		if idx1, isExist := rec[n]; isExist {
			return []int{idx1 + 1, i + 1}
		}
		rec[target-n] = i
	}
	return nil
}
